stream_alert/rule_processor/payload.py

```diff
@@ -0,0 +1,328 @@
+        return
+
+    return payload_map[service](raw_record=raw_record, entity=entity)
+
+
+class StreamPayload(object):
+    """Container class for the StreamAlert payload object.
+
+    Attributes:
+        raw_record: The record from the AWS Lambda Records dictionary.
+
+        valid: A boolean representing if the record is deemed valid by
+            parsing and classification.
+
+        service: The aws service where the record originated from. Can be
+            either S3 or kinesis.
+
+        entity: The instance of the sending service. Can be either a
+            specific kinesis stream or S3 bucket name.
+
+        log_source: The name of the logging application which the data
+            originated from.  This could be osquery, auditd, etc.
+
+        type: The data type of the record - json, csv, syslog, etc.
+
+        record: A list of parsed and typed record(s).
+
+    Public Methods:
+        refresh_record
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, **kwargs):
+        """
+        Keyword Args:
+            raw_record (dict): The record to be parsed - in AWS event format
+        """
+        self.raw_record = kwargs['raw_record']
+        self.entity = kwargs['entity']
+
+        self.pre_parsed_record = None
+        self.type = None
+        self.log_source = None
+        self.records = None
+        self.valid = False
+
+    def __repr__(self):
+        repr_str = ('<{} valid:{} log_source:{} entity:{} type:{} '
+                    'record:{}>').format(self.__class__.__name__, self.valid,
+                                         self.log_source, self.entity,
+                                         self.type, self.records)
+
+        return repr_str
+
+    @abstractproperty
+    def service(self):
+        """Read only service property enforced on subclasses.
+
+        Returns:
+            [string] The service name for this payload type.
+        """
+
+    @abstractmethod
+    def pre_parse(self):
+        """Pre-parsing method that should be implemented by all subclasses.
+        This establishes the `pre_parsed_record` property to allow for parsing.
+
+        Returns:
+            [generator] Yields instances of `self` back to the caller with the
+                proper `pre_parsed_record` set. Conforming to the interface of
+                returning a generator provides the ability to support multi-record
+                payloads, such as those similar to S3.
+        """
+
+    def refresh_record(self, new_record):
+        """Replace the currently loaded record with a new one.
+
+        Used mainly when S3 is used as a source, due to looping over files
+        downloadd from S3 events versus all records being readily available
+        from a Kinesis stream.
+
+        Args:
+            new_record [string]: A new raw record to be parsed
+        """
+        self.pre_parsed_record = new_record
+        self.type = None
```

Comment:
these initializations should be encapsulated into a method, it's the same thing as in `__init__`